Download transfers starting from latest block header (#1467)
status-im/status-go: services/wallet/README.md

Wallet API
==========

Wallet service starts a loop that watches for new transfers (eth and erc20).

> To correctly start a service two values need to be changed in the config:

Reviewer: ```suggestion
To correctly start the service, two values need to be changed in the config:
```

1. Set Enable to true in WalletConfig

```json
{
  "WalletConfig": {
    "Enabled": true,
  }
}
```

2. And expose wallet API with APIModules

```
{
  APIModules: "eth,net,web3,peer,wallet",
}
```

> Wallet API

Reviewer: Again Wallet API section?

      "input": "0x",
      "v": "0xa95",
      "r": "0xd171c582139765f44fa1401edcdd377ca1b350ee10c0685073a5d470fc3625c6",
      "s": "0x3392d24da2a13449345fbe8210d5af89f01ee21b456dc3c226e7cd7a0509fed",
      "hash": "0x56be9c3cf40243e62428a203e37ce2104cfa12bef7d9e47988fd769617d361c6"
    },
    "Receipt": {
      "root": "0x",
      "status": "0x1",
      "cumulativeGasUsed": "0x5208",
      "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
      "logs": [],
      "transactionHash": "0x56be9c3cf40243e62428a203e37ce2104cfa12bef7d9e47988fd769617d361c6",
      "contractAddress": "0x0000000000000000000000000000000000000000",
      "gasUsed": "0x5208"
    }
  }
]
```

Signals
-------

> Two signals will are emmited:

Reviewer: ```suggestion
Two signals will are emitted:
```

> 1. Signal `newblock`

Reviewer: ```suggestion
1. `newblock` signal
```

> Emmited when transfers from new block were added to the database. In this case block number if the number of this new block.

Reviewer: ```suggestion
Emitted when transfers from a new block were added to the database. In this case the block number is the number of this new block.
```

```json
{
  "type": "wallet",
  "event": {
    "Type": "newblock",
    "BlockNumber": 10
  }
}
```

> 2. Signal `reorg`.

Reviewer: ```suggestion
2. `reorg` signal
```

> Emmited when part of blocks were removed. Starting from a given block number all transfers were removed.

Reviewer: ```suggestion
Emitted when part of blocks were removed. Starting from a given block number, all transfers were removed.
```

Reviewer 2: We should add information what is an expected action after each signal. Whether a call to `wallet_getTransfers` should happen and with what params from received signal.

Author: that was pretty much the whole idea. added some details to readme
